File: dart/lib/model/v3_siri_line_ref.dart
part of swagger.api;

class V3SiriLineRef {
  /* Siri LineRef */
  String lineRef = null;
/* Siri DirectionRef  (in, out, up, down, clockwise, counterclockwise, Inbound, Outbound) */
  int directionRef = null;
  //enum directionRefEnum {  1,  2,  5,  10,  16,  32,  65,  130,  };

  V3SiriLineRef();

  @override
  String toString() {
    return 'V3SiriLineRef[lineRef=$lineRef, directionRef=$directionRef, ]';
  }

  V3SiriLineRef.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    lineRef = json['line_ref'];
    directionRef = json['direction_ref'];
  }

  Map<String, dynamic> toJson() {
    return {
      'line_ref': lineRef,
      'direction_ref': directionRef
     };
  }

  static List<V3SiriLineRef> listFromJson(List<dynamic> json) {
    return json == null ? new List<V3SiriLineRef>() : json.map((value) => new V3SiriLineRef.fromJson(value)).toList();
  }

  static Map<String, V3SiriLineRef> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, V3SiriLineRef>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new V3SiriLineRef.fromJson(value));
    }
    return map;
  }
}

File: csharp-dotnet2/src/main/CsharpDotNet2/IO/Swagger/Model/V3SiriLineRef.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class V3SiriLineRef {
    /// <summary>
    /// Siri LineRef
    /// </summary>
    /// <value>Siri LineRef</value>
    [DataMember(Name="line_ref", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "line_ref")]
    public string LineRef { get; set; }

    /// <summary>
    /// Siri DirectionRef  (in, out, up, down, clockwise, counterclockwise, Inbound, Outbound)
    /// </summary>
    /// <value>Siri DirectionRef  (in, out, up, down, clockwise, counterclockwise, Inbound, Outbound)</value>
    [DataMember(Name="direction_ref", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "direction_ref")]
    public int? DirectionRef { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class V3SiriLineRef {\n");
      sb.Append("  LineRef: ").Append(LineRef).Append("\n");
      sb.Append("  DirectionRef: ").Append(DirectionRef).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
